### PR TITLE
Redis

### PR DESCRIPTION
**Pull request by usama-asfar**
I have created a simple todo app with express, mongoose and Redis. This might be so helpful and give a brief idea for people who are trying to integrate Redis on their mongoose DB.
Thank You.

### CODE REVIEW

File: examples/redis-todo/db/models/userModel.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const { Schema, model } = require('mongoose')
+const jwt = require('jsonwebtoken')
+const bcrypt = require('bcryptjs')
+
+const userSchema = new Schema({
+  name: { type: String, required: true },
+  username: { type: String, unique: false, required: true }, // TODO: change this to true
```
Comment (vkarpov15) [inline examples/redis-todo/db/models/userModel.js]: Why is this TODO here?

File: examples/redis-todo/db/models/userModel.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  email: { type: String, unique: false, required: true },
+  password: { type: String, required: true }
+}, { timestamps: true, versionKey: false })
+
+userSchema.methods.toJSON = function () {
+  const user = this.toObject() // this = user
+  delete user.password
```
Comment (vkarpov15) [inline examples/redis-todo/db/models/userModel.js]: Generally speaking, I'd prefer if `password` were stored outside the user schema. Here's some more thoughts: https://thecodebarbarian.com/thoughts-on-user-passwords-in-rest-apis

File: examples/redis-todo/middleware/auth.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    const authToken = req.header('x-auth')
+    if (!authToken) return res.status(404).send({ msg: 'AuthToken not found' })
+
+    const decodedValue = jwt.verify(authToken, 'token')
+    if (!decodedValue) return res.status(401).send({ msg: 'Invalid Authentication' })
+
+    req.userId = decodedValue.userId
+    next()
+  } catch (err) {
+    res.status(401).send({ msg: 'Invalid Authenication', err })
```
Comment (vkarpov15) [inline examples/redis-todo/middleware/auth.js]: Typo in 'authentication'

File: examples/redis-todo/db/models/userModel.js (hunk 1 above)
Comment (usama-asfar) [reply examples/redis-todo/db/models/userModel.js]: to change unique to true,

Comment (usama-asfar) [reply examples/redis-todo/db/models/userModel.js]: fixed the issue

File: examples/redis-todo/db/models/userModel.js (hunk 2 above)
Comment (usama-asfar) [reply examples/redis-todo/db/models/userModel.js]: I had separated password from user schema to new password schema

File: examples/redis-todo/middleware/auth.js (hunk 3 above)
Comment (usama-asfar) [reply examples/redis-todo/middleware/auth.js]: fixed the issue

**Pull request merged**